### PR TITLE
Feature/use redux

### PR DESCRIPTION
**Pull request by jaferIdrees**
### Project requirements

- First of all - make sure that you keep state immutable in reducers!
- Secondly, change the default state in your books reducer from an empty array to an array with a few books. One of the goals is to display them in your React components.
- Import Redux Provider and your store in the main component of your React app.
- Connect your app to the Redux store.
- Make sure that list of books displays books from Redux store.
- Use useSelector to consume the state.
- In your React component responsible for adding new books, add functionality for "Add new" button click:
- Import Redux useDispatch.
- Set the data inputs in the local React state (set title and author - NOTE: categories and comments are NOT part of this step).
- Takes the inputs from the state and generate a unique id and add them to an object.
- Once your new book object is ready to be submitted to Redux store, dispatch a corresponding action (import Action Creator that you created before and use it here).
- In your React component responsible for removing books, add a functionality for "Remove" button click:
- Import Redux useDispatch.
- Dispatch a corresponding action (import Action Creator that you created before and use it here).
- In your React component responsible for categories, add a functionality for "Check status" button click:
- Import Redux useDispatch.
- Dispatch a corresponding action (import Action Creator that you created before and use it here).

### CODE REVIEW

File: src/components/AddBookForm.js [hunk 1]
```diff
@@ -1,11 +1,21 @@
 import React from 'react';
+import { useDispatch } from 'react-redux';
+import { addBook } from '../redux/books/books';
 
 function AddBookForm() {
```
Comment (FrederickMih) [inline src/components/AddBookForm.js:6]: - [x] Hi @jaferIdrees, based on the requirements of this project, this component should contain a local state with title and author. They should have their inputs that update the local state every time they update(onChange). I think you have the option of using the useState Hook when creating functional components as in your case or we can use React ES6 classes which come with a constructor function to instantiate the initial requirements of the components

File: src/components/AddBookForm.js [hunk 2]
```diff
@@ -1,11 +1,21 @@
...
+  const dispatch = useDispatch();
+  const handleSubmit = (e) => {
```
Comment (FrederickMih) [inline src/components/AddBookForm.js:15]: - [x] Fantastic job :+1: for implementing this function but let's keep in mind that based on the requirements of this project this function(handleSubmit) takes the inputs from the state and `generate a unique id` :red_circle:  and adds them to an object then dispatches the `addBook` action creator :+1: with the object as its argument. Therefore, kindly ensure that the unique id is generated as requested.

File: src/components/Categories.js [hunk 3]
```diff
@@ -1,8 +1,17 @@
...
 function Categories() {
+  const state = useSelector((state) => state.categoriesReducer);
+  const dispatch = useDispatch();
   return (
-    <button type="button">Check status</button>
+    <div>
+      <h2>{state}</h2>
+      {' '}
+
+      <button type="button" onClick={() => dispatch(checkStatus())}>Check status</button>
```
Comment (FrederickMih) [inline src/components/Categories.js]: - [ ] Based on the requirements of this project, this button should use a function `checkCategoriesStatus`(named as you wish) made in this component. The functions should dispatch the `checkStatus` action creator. Let's kindly implement this function inside this component.

**Review by FrederickMih — CHANGES_REQUESTED** [review]
Hi @jaferIdrees , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


# To Highlights

- Github workflow has been followed :heavy_check_mark: 
- PR has a title and summary :heavy_check_mark: 
- No Linter issue detected :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/AddBookForm.js [hunk 4]
```diff
@@ -1,11 +1,36 @@
...
+    e.target.reset();
+  };
   return (
-    <form>
+    <form onSubmit={handleSubmit}>
       <h2>ADD NEW BOOK</h2>
-      <input type="text" id="titleInput" placeholder="Book title" />
-      <input type="text" id="authorInput" placeholder="Author" />
+      <input type="text" id="titleInput" required name="title" placeholder="Book title" value={titleEntry} />
+      <input type="text" id="authorInput" required name="author" placeholder="Author" value={authorEntry} />
```
Comment (DammyShittu) [inline src/components/AddBookForm.js]: - [ ] You are yet to implement the use of local states for title and author as suggested. While it appears your inputs are defined by the bookData state, you seem to dispatch the input values directly which defeats the purpose of using the local state. You are expected to dispatch the state.

- [ ] You are also expected to set the value of the inputs using onChange

  If you are struggling with the local states and dispatch you can look at these articles where the onChange method is used properly
  -  [https://www.genui.com/resources/react-hooks-simplify](https://www.genui.com/resources/react-hooks-simplify)
  - [https://dmitripavlutin.com/controlled-inputs-using-react-hooks/](https://dmitripavlutin.com/controlled-inputs-using-react-hooks/)

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @jaferIdrees,

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/AddBookForm.js (hunk 4 above)
Comment (jaferIdrees) [reply src/components/AddBookForm.js]: Hi @DammyShittu.
Many thanks for the advice 👍

**Review by topeogunleye — APPROVED** [review]
### Hi @jaferldrees, 

# Status: Approved ✔️
Excellent job on making the changes requested by the previous code reviewer. Keep it up 👏

Your project is complete! There is nothing else to say other than... it's time to merge it 💪
Congratulations! 🎉
congratulations
>
## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Followed Github flow :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title :heavy_check_mark:
 
## [Optional] suggestions:
* Nothing to mention.
 
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
 
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @topeogunleye in your question so I can receive the notification.



------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request closed without merging**